### ISSUE TITLE
Passim causing a large delay if Avahi is not enabled

### ISSUE THREAD

**Issue by michaellass** [hughsie/passim#15]
Not sure if this should be considered a bug but it is something I noticed after upgrading to fwupd 1.9.5:

On systems without a Avahi being enabled via systemd, passim causes large delays when invoked:
```
% time passim --version
Failed to connect to daemon: Error calling StartServiceByName for org.freedesktop.Passim: Failed to activate service 'org.freedesktop.Passim': timed out (service_start_timeout=25000ms)passim --version  0,01s user 0,00s system 0% cpu 25,026 total
```

`org.freedesktop.Passim.service` tries to start `passimd` which fails without Avahi:
```
% sudo /usr/lib/passimd
(process:27075): GLib-DEBUG: 13:57:03.729: setenv()/putenv() are not thread-safe and should not be used after threads are created
** (passimd:27075): DEBUG: 13:57:03.733: loading resources from /var/lib/passim/data
(passimd:27075): GLib-GIO-DEBUG: 13:57:03.733: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)

** (passimd:27075): WARNING **: 13:57:03.736: failed to contact daemon: failed to contact Avahi: Error calling StartServiceByName for org.freedesktop.Avahi: Unit dbus-org.freedesktop.Avahi.service not found.
```

`passimd` itself quits very quickly in this case, so I'm not exactly sure why `org.freedesktop.Passim.service` does not fail quickly, too, but instead runs into the 25 second timeout.

The consequence for `fwupdmgr` is that the automatic start of `fwupd.service` takes too long as well. On second invocation, fwupd.service is then already running and everything is fine:
```
% fwupdmgr refresh --force   
Failed to connect to daemon: Error calling StartServiceByName for org.freedesktop.fwupd: Timeout was reached
% fwupdmgr refresh --force
Updating lvfs
Downloading?             [     \                                 ]
Successfully downloaded new metadata: 7 local devices supported
```

Avahi is not supposed to be activated via dbus if systemd is present, hence it must be enabled via systemd to fix the issue described above: https://github.com/lathiat/avahi/blob/006200d5ec85b3e6f715136a10dd23e01b1dea4b/avahi-daemon/org.freedesktop.Avahi.service#L22

Enabling Avahi via systemd will create a link `/etc/systemd/system/dbus-org.freedesktop.Avahi.service` which then fixes the automatic activation of Avahi when starting `org.freedesktop.Passim.service`.

Maybe there is a way to let `org.freedesktop.Passim.service` timeout much more quickly or fail directly when `passimd` terminates? Otherwise, this behavior may need documentation so that users enable Avahi via systemd when they want to use fwupd.

**Comment by hughsie** [comment]
Is there w a way to change the service_start_timeout=25000ms in passim.service? e.g. say 2000ms?

Or, maybe PassimClient should look for Avahi, and if not running then don't autostart passimd?

**Comment by michaellass** [comment]
I only now looked into `/usr/lib/systemd/system/passim.service`. The following change also fixes the issue for me:
```diff
--- /usr/lib/systemd/system/passim.service
+++ /usr/lib/systemd/system/passim.service
@@ -3,6 +3,7 @@
 Documentation=https://github.com/hughsie/passim
 After=dbus.service
 Before=display-manager.service
+Wants=avahi-daemon.service
 
 [Service]
 Type=dbus
```

The root cause of this is that in `/usr/share/dbus-1/system-services/org.freedesktop.Avahi.service` the underlying service for Avahi is given as `SystemdService=dbus-org.freedesktop.Avahi.service` but `dbus-org.freedesktop.Avahi.service` only exists if Avahi was enabled via systemd. Otherwise it is only known as `avahi-daemon.service`.

This would easily be fixable in Avahi. I found an issue report on this that is nearly 8 years old... https://github.com/lathiat/avahi/issues/29 It could of course be that this is the desired behavior so that Avahi is not activated without the administrator knowing.

A short timeout would be OK as well. However, I don't know where the 25 seconds are coming from. passim.service here says `TimeoutSec=180`. Also, starting `passim.service` manually immediately fails when Avahi is not available. So I think the 25 seconds come from somewhere in dbus waiting for the service to come up.
